packages/transaction-controller/src/utils/validation.ts

```diff
@@ -247,6 +247,43 @@ export function validateParamTo(to?: string) {
   }
 }
 
+/**
+ * Validates a transaction batch request.
+ *
+ * @param options - Options bag.
+ * @param options.internalAccounts - The internal accounts added to the wallet.
+ * @param options.request - The batch request object.
+ */
+export function validateBatchRequest({
+  internalAccounts,
+  request,
+}: {
+  internalAccounts: string[];
+  request: TransactionBatchRequest;
+}) {
+  const { origin } = request;
+  const isExternal = origin && origin !== ORIGIN_METAMASK;
+
+  const transactionTargetsNormalized = request.transactions.map((tx) =>
+    tx.params.to?.toLowerCase(),
+  );
+
+  const internalAccountsNormalized = internalAccounts.map((account) =>
+    account.toLowerCase(),
+  );
+
+  if (
+    isExternal &&
+    transactionTargetsNormalized.some((target) =>
+      internalAccountsNormalized.includes(target as string),
+    )
+  ) {
+    throw rpcErrors.invalidParams(
+      'External transactions to internal accounts are not supported',
```

Comment:
I'm not sure if this internal vs external transaction nomenclature exists already on our dev docs, but if not I think it can be confusing or not explicit enough in this error message.

We (internally within the team) usually distinguish these by calling them dApp initiated versus wallet initiated transactions, and I think that's more explicit.